data_structures/stack.h

```diff
@@ -109,16 +115,16 @@ class stack {
     }
 
     /** Clear stack */
-    void clear() { stackTop = nullptr; }
+    void clear() {
+        deleteAllNodes();
+        size = 0;
+    }
 
     /** Overload "=" the assignment operator */
     stack<Type> &operator=(const stack<Type> &otherStack) {
         node<Type> *newNode, *current, *last;
 
-        /* If stack is no empty, make it empty */
-        if (stackTop != nullptr) {
-            stackTop = nullptr;
-        }
+        deleteAllNodes();
```

Comment:
from here on until line 147 the size of _this_ is not accurate anymore. What about resetting the size within deleteAllNodes?

Reply:
in this case, line 120 in clear() would not be needed anymore, as well.

Reply:
I do not see the variable `size` to be _really_ used anywhere (well the method `display()` reads it, note that the `size` could be computed there; there is nothing like `get_size()` - having it would justify presence of the `size` member). I think that this is beyond the scope of this PR.

Reply:
Yes, I'm fine with that.